pkgs/os-specific/linux/nvidia-x11/settings.nix

```diff
@@ -45,13 +45,22 @@ let
       "OUTPUTDIR=." # src/libXNVCtrl
     ];
 
+    patches = [
+      # Patch the Makefile to also produce a shared library.
+      (if (lib.versions.major nvidia_x11.settingsVersion) == "470" then ./libxnvctrl-build-shared-470.patch
+      else if (lib.versions.major nvidia_x11.settingsVersion) == "390" then ./libxnvctrl-build-shared-390.patch
+      else if (lib.versions.major nvidia_x11.settingsVersion) == "340" then ./libxnvctrl-build-shared-340.patch
```

Comment:
Function application has a higher precedence than equality so there's no need for parenthesis around the calls to `lib.versions.major`.

Reply:
Fixed.